docs/content/sips/xxx-spin-state-wasi.md

```diff
@@ -0,0 +1,131 @@
+
+Created: June 7th, 2022
+
+Updated: June 8th, 2022
+
+## Background
+
+Apart from reading the file system of the host machine, there's no way to read
+logs or manifest of a running Spin app. This means users are limited in what
+they know about a Spin app while it's running, unless they have access to the
+environment it is on.
+
+[Laundromat](https://github.com/ecumene/laundromat) is a tool for reading logs
+and organizing them in a quicker format for developer velocity, and it gets
+around this by serving the logs of a Spin app as a REST API. This is limited
+because
+
+- it must have access to the file system which users may not have access to
+- being a server itself it must be forwarded out of whatever network users are
+  running it in, along with Spin itself
+- the file system is itself limited as new logs won't notify the user.
+  system file events could be used but they're notoriously platform dependent.
+
+If Spin exposed a "monitoring filesystem" that was accessible by WASI FS we
+could open the door to many hypervisor-like tools.
+
+### The `~/.spin/` directory
+
+The `~/.spin/` directory is where Spin keeps logs and other ephemeral data for
+apps.
+
+```
+tree ~/.spin/
+└── appname
+    ├── key-value
+    └── logs
+        ├── componentname_stderr.txt
+        └── componentname_stdout.txt
+```
+
+## Proposal
+
+WASI provides a system interface where components can access files as if they
+were system files in their program, except those files don't have to be on disk
+and don't even have to be files at all.
+
+### `~/.spin/` 2: Escape 2 Madigascar
+
+Exposing `~/.spin/` to components doesn't work because all files and folders
+remain in the same state they were mounted at. Any new logs won't be included
+when read again.
+
+If instead of mounting a `~/.spin/` directory on the host machine, we allowed
+users to mount a directory internal to Spin in-memory this would be updated
+on subsequent reads while having the access control of a mounted directory.
+
+#### Mount: `/monitor/`
+
+Inside each app folder are the stdout/stderr for each component. This is what we
+want to make available as a file system for guest modules. To do so they need to
+be explicitly mounted. However since we don't want to mount the actual
+file system behind Spin because it's ephemeral and temporary, we should be
+explicit about how this file system is different from the host file system.
+
+> **_MITCH:_** I am very iffy about this
+
+> A. `spin_source = "/monitor/logs"` is verbose
+
+> B. `spin_source` is confusing unless you grok'd the docs for it
+
+```diff
+-files = [{ source = "~/.spin/[...]/logs", destination = "/logs" }]
++files = [{ spin_source = "/monitor/logs/", destination = "/monitor/logs" }]
+```
+
+The `/monitor` path is a special path in Spin where logs are accessed from. If
+users provide this path to their components under the `spin_source` source key
+then the component's logs would be accessible to the guest module.
```

Comment:
Rather than making this a magic mount point, I think we can add explicit configuration for "enable the monitor interface, please". Sort of like the existing `allowed_http_hosts` configures the outbound-http interface. Something like `spin_monitoring = true` on the component.